### PR TITLE
Hel 334 hel 340 photo size handling

### PR DESCRIPTION
**Pull request by jannetalvio**
HEL-334 HEL-340.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Kuvaselaamo branch is deployed to platta: https://kuvaselaamo-pr298.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Kuvaselaamo branch is deployed to platta: https://kuvaselaamo-pr298.dev.hel.ninja** :rocket::rocket::rocket:

**Review by HeikkM — APPROVED** [review]

File: hkm/finna.py [hunk 1]
```diff
@@ -234,10 +235,24 @@ def _get_finna_result(self, url, payload):
 
         return result_data
 
+    def get_thumbnail_image_url(self, record):
+        if identifier := self.get_identifier(record):
```
Comment (nikomakela) [inline hkm/finna.py:239]: oh, I didn't know that we have a new enough Python for walrus here! :)

File: hkm/finna.py [hunk 2]
```diff
@@ -299,5 +314,20 @@ def get_labeled_high_res_url(self, record, label):
...
+            return None
+
+        LOG.debug(
+            "Found images_extended",
+            extra={"data": {"result_data": images_extended}},
+        )
+
+        for image_extended in images_extended:
+            if identifier := image_extended.get("identifier"):
+                return identifier
```
Comment (nikomakela) [inline hkm/finna.py:328]: This return from inside a loop looks wierd, but I think it's totally acceptable in Python world 👌

File: package.json [hunk 3]
```diff
@@ -2,8 +2,8 @@
   "name": "hkm",
   "version": "0.0.1",
   "scripts": {
-    "browser-test": "testcafe \"chrome --window-size='1920,1080'\" browser-tests/enable-dotenv.js browser-tests/ --live",
-    "browser-test:ci": "testcafe \"chrome:headless --disable-gpu --window-size='1920,1080'\" --screenshots path=report takeOnFails=true --video report browser-tests/"
+    "browser-test": "testcafe \"chrome --lang=fi-FI --window-size='1920,1080'\" browser-tests/enable-dotenv.js browser-tests/ --live",
```
Comment (nikomakela) [inline package.json:5]: Why are these 2 using a different language? Shouldn't they be exactly the same and only difference be that another one is ran in a non graphical environment.

**Review by nikomakela — CHANGES_REQUESTED** [review]
The 2 browser testing commands are using different language! They shouldn't because the scripts are the same.

File: browser-tests/utils/route.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import settings from "./settings";
 
 export function getRoute(path: string) {
-  return `${settings.baseUrl}${path}`;
+  return `${settings.baseUrl}/language/?lang=en&next=${path}`;
```
[the remark this answers is not in the document]
Comment (nikomakela) [reply browser-tests/utils/route.ts:4]: Yea, the language could at least be parametrized.

[the remark this answers is not in the document]
Comment (jannetalvio) [reply browser-tests/utils/route.ts:4]: This is actually the functional fix to the test issues failing depending on where they are run. This uses the application functionality to force language to be English before each test. And all the tests assume that tested language is English, thus I wouldn't really see a problem having the language hard coded, as all the English UI references are hard coded as well.

File: package.json (hunk 3 above)
Comment (jannetalvio) [reply package.json:5]: This is actually a mistake. Forcing the UI language is necessary to make sure test are run in the language (English) that the tests assume. But the ci configuration had no effect in the ci-env and the local language was there just to simulate similar failure than in ci-env. Will remove both. Forcing English as the UI language is done using the app feature to set the language before page load.

**Review by nikomakela — APPROVED** [review]

**Comment by jannetalvio** [comment]
Closed without merging,

**Comment by terovirtanen** [comment]
**Kuvaselaamo branch is deployed to platta: https://kuvaselaamo-pr298.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**